pkgs/tools/virtualization/nixos-container/nixos-container.pl

```diff
@@ -272,6 +273,7 @@ sub isContainerRunning {
 
 sub terminateContainer {
     my $leader = getLeader;
+    system("systemctl", "stop", "--no-block", "container\@$containerName");
     system("machinectl", "terminate", $containerName) == 0
```

Comment:
Urgh, this feels like a hack. Doesn't `machinectl terminate` by itself make the systemd unit become stopped?

Reply:
this is consequence of removing `--keep-unit`, I haven't found a better way to stop unit with machine.

`machinectl stop` does stop unit fine. It is because machine exits with normal result. But when it runs `terminate`, it exits with failure result, and systemd restarts unit due to `Restart=on-failure` policy.